core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -215,14 +227,115 @@ public boolean isAvailable() {
       return file.exists();
    }
 
+   private boolean isCorruptData() throws IOException {
+      ByteBuffer buf = ByteBuffer.allocate(KEY_POS_LATEST);
+      long pos = filePos;
+      buf = readChannel(buf, pos, KEY_POS_LATEST);
+      if (buf.remaining() > 0)
+         return false;
+      buf.flip();
+      FileEntry fe = new FileEntry(pos, buf);
+      // Explicitly calculate the correct size as incorrect value stored in fe.size
+      int actualSize = KEY_POS_LATEST + fe.keyLen + fe.dataLen + fe.metadataLen + fe.internalMetadataLen;
+
+      // Attempt to read the next entry in the file
+      pos += actualSize;
+      buf = readChannel(buf, pos, KEY_POS_LATEST);
+
+      // Only single FileEntry present with correct size, so data is not corrupt
+      if (buf.remaining() > 0)
+         return false;
+
+      buf.flip();
+      fe = new FileEntry(pos, buf);
+      channel.position(filePos);
+      return fe.size == 0;
```

Comment:
This detection is still wrong and is the first issue with `sfs/corrupt/private-metadata-only.dat`

The fe contents are:
```
fe = {SingleFileStore$FileEntry@6483} "FileEntry@110{size=4194304, actual=3473435}"
 offset = 110
 size = 4194304
 keyLen = 1310720
 dataLen = 1441792
 metadataLen = 0
 internalMetadataLen = 720895
 expiryTime = -26623
```

I am thinking we should probably have the sanity check if the size = keyLen + dataLen + metadataLen + internalMetadataLen + 28. I *think* we also have to add in 8 if expiryTime != -1 as well.